src/main/java/org/libtorrent4j/SessionManager.java

```diff
@@ -622,13 +628,13 @@ public void remove(TorrentHandle th) {
      * @param timeout in seconds
      * @return the bencoded info or null
      */
-    public byte[] fetchMagnet(String uri, int timeout) {
+    public byte[] fetchMagnet(String uri, int timeout, File tempDir) {
```

Comment:
Codacy found an issue: [The method 'fetchMagnet(String, int, File)' has an NPath complexity of 816, current threshold is 200](https://app.codacy.com/manual/aldenml/libtorrent4j/pullRequest?prid=5876791)